plugins/outputs/cloud_pubsub/cloud_pubsub.go

```diff
@@ -193,8 +205,20 @@ func (ps *PubSub) toMessages(metrics []telegraf.Metric) ([]*pubsub.Message, erro
 			b = []byte(encoded)
 		}
 
+		b, err = ps.encoder.Encode(b)
+		if err != nil {
+			ps.Log.Debugf("could not encode metric: %v", err)
+			continue
+		}
```

Comment:
If we cannot encode, then we drop those metrics on the floor?

Reply:
I'm wondering if this should be an error rather than a hidden message unless debug is enabled. We typically will error if we fail to serialize or parse something.

I might leave this for @srebhan for next week when he gets back and we can wrap up the other items here.

Reply:
I agree, we should not silently drop metrics! Please convert this to `ps.Log.Errorf`.